### ISSUE TITLE
Allow keywords arguments for roots

### ISSUE THREAD

**Issue by Kolaru** [JuliaIntervals/IntervalRootFinding.jl#116]
I new issue found while writing the documentation. The current situation is awkward.

I think we could go with the following scheme:

**Positional arguments**
1. The function.
2. The search region.
3. The method.

Optionally for Krawczyk and Newton:
1. The function.
2. Its derivative.
3. The search region.
4. The method.

**Keyword arguments**
- The type of search (once #97 is merged).
- The tolerance (both absolute and relative once #66 is fixed).
- The maximum number of intervals considered (once #59 is implemented).

Maybe the method should be both a keyword and an optional argument, not sure about this one.

**Comment by dpsanders** [comment]
I think it might be time to introduce a `RootsProblem` type, similar to `ODEProblem` in `DifferentialEquations`. 

This would contain all of the parameters of the search etc.
There could still be a convenience `roots` function for quick usage, or just create the `RootsProblem` object and call `roots` on that.

**Comment by Kolaru** [comment]
Indeed, as I add things to the list, the more I agree with the idea of a `RootsProblem`, this will probably simplify passing around the parameters internally.

It may also be good to force keywords parameters to be passed through such object and have `roots` with only positional arguments (i.e. have it as a convenient function for simple cases).

**Comment by dpsanders** [comment]
Eg. See how it's done in MDBP.jl:

https://discourse.julialang.org/t/ann-mdbm-jl-a-root-finding-package-for-system-of-non-linear-equations/22141

It would maybe be useful to be able to use either package with the same or similar syntax.

**Comment by Kolaru** [comment]
It would be good to follow the common interface from https://github.com/SciML/CommonSolve.jl

**Comment by lucaferranti** [comment]
I use that in IntervalLinearAlgebra.jl. it's very nice

**Comment by Kolaru** [comment]
#203 change the handling of the keyword argument to be much nicer.

Following CommonSolve.jl convention is a separate issue.
